web3/exceptions.py

```diff
@@ -20,15 +35,15 @@ class BadFunctionCallOutput(Exception):
     pass
 
 
-class BlockNumberOutofRange(Exception):
+class BlockNumberOutofRange(Web3Exception):
     """
     block_identifier passed does not match known block.
     """
 
     pass
 
 
-class CannotHandleRequest(Exception):
+class CannotHandleRequest(Web3Exception):
```

Comment:
Are `TooManyRequests` and `MultipleFailedRequests` left inheriting form `Exception` intentionally?

Reply:
🤔 Nope, not sure how I missed those. Nice catch, thanks!